### PR TITLE
BPR-5076 - business partner search in role assignment editor limited to user-all…

### PR DESCRIPTION
**Pull request by ElaCzapiewska**
business partner search in role assignment editor limited to user-allowed BPs

### CODE REVIEW

File: src/server/routes/index.js [hunk 1]
```diff
@@ -695,6 +698,26 @@ module.exports.deleteAssignableRole = async function(req, res)
...
+
+        const serviceClient = req.opuscapita.serviceClient;
+        const userRoles = req.opuscapita.userData('roles');
+        const userBusinesPartnerIds = await req.opuscapita.getUserBusinessPartnerIds();
+        
+        const businessPartnerApi = new BusinessPartner(serviceClient, userRoles, userBusinesPartnerIds);
+        return res.json(await businessPartnerApi.getBusinessPartners(name));
+  
+    }
+    catch(err) {
```
Comment (jaconsta) [inline src/server/routes/index.js:716]: is this catch required? The function is wrapped by a `catchError` middleware at L104

File: src/client/api/User.js [hunk 2]
```diff
@@ -80,7 +80,13 @@ class User extends ApiBase
     deleteRole(roleId)
     {
         return this.ajax.delete(`/user/api/roles/${roleId}`).then(res => res.body).catch(this.getErrorFromResponse);
+    }  
+    
+    getUserBusinessPartners(name)
+    {
+        return this.ajax.get('/user/api/getUserBusinessPartners').query({ name }).then(res => res.body).catch(this.getErrorFromResponse);
     }
+
```
Comment (jaconsta) [inline src/client/api/User.js:89]: Please, remove the empty line

File: src/server/acl.json [hunk 3]
```diff
@@ -63,6 +63,11 @@
                 "type": "rest",
                 "resourceId": "^/user/substitutionPeriodUserEditor",
                 "actions": ["view"]
+            },
+            {
```
Comment (jaconsta) [inline src/server/acl.json:67]: Would be nice if only the correct resource groups are assigned to the new endpoint.

**Review by jaconsta — CHANGES_REQUESTED** [review]
Really nice work! I left some aesthetics changes.

File: src/server/api/BusinessPartner.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+     *
+     * @returns Array<BusinessPartnerObject>
+     */
+    async getBusinessPartners(name)
+    {
+        try {
+            const businessPartnerIds = 
+                this.userRoles.includes('admin') 
+                ? ''
+                : '&id=' + this.userBusinessPartnerIds.join(',');
```
[the remark this answers is not in the document]
Comment (ElaCzapiewska) [reply src/server/api/BusinessPartner.js]: ok, changed

File: src/server/api/BusinessPartner.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                    'business-partner', 
+                    `/api/business-partners?name=${encodeURIComponent(name)}${businessPartnerIds}&pageSize=100&page=1`, true);      
+            
+            return businessPartners[0];
+        }
+        catch(err) {
+            return [];
+        }
+    } 
+}
```
[the remark this answers is not in the document]
Comment (ElaCzapiewska) [reply src/server/api/BusinessPartner.js]: ok, changed

File: src/server/routes/index.js [hunk 6]
```diff
@@ -9,6 +9,8 @@ const UserToRoles = require('../api/UserToRoles.js');
 const userToUserRelations = require('./userToUserRelations');
 const substitutionPeriods = require('./substitutionPeriod.js');
 
+const BusinessPartner = require('../api/BusinessPartner.js');
```
[the remark this answers is not in the document]
Comment (ElaCzapiewska) [reply src/server/routes/index.js]: ok, changed, I move it upper, just under another recall of ../api/xx

File: src/server/routes/index.js (hunk 1 above)
Comment (ElaCzapiewska) [reply src/server/routes/index.js]: right, corrected

File: src/server/acl.json (hunk 3 above)
Comment (ElaCzapiewska) [reply src/server/acl.json]: the new endpoint is now available only for assignable-role-mgmt, role-mgmt, x-tenant-user-mgmt resource groups. That means admin, customer-admin and supplier-admin can access it but user without one of these roles has no access to the endpoint

**Review by jaconsta — APPROVED** [review]
:rocket:  Super! please Squash and merge :)

**Pull request merged**